src/module.c

```diff
@@ -691,7 +691,7 @@ JL_DLLEXPORT void jl_checked_assignment(jl_binding_t *b, jl_value_t *rhs)
                 jl_errorf("invalid redefinition of constant %s",
                           jl_symbol_name(b->name));
             }
-            jl_printf(JL_STDERR, "WARNING: redefining constant %s\n",
+            jl_printf(JL_STDERR, "WARNING: redefinition of constant %s may cause Julia to fail\n",
```

Comment:
```suggestion
            jl_printf(JL_STDERR, "WARNING: redefinition of constant %s may cause the program to fail\n",
```

Reply:
```suggestion
            jl_printf(JL_STDERR, "WARNING: redefinition of constant %s may cause problems (including crashes), continue at your own risk\n",
```

Reply:
Or:
```suggestion
            jl_printf(JL_STDERR, "WARNING: redefinition of constant %s may sometimes fail or cause crashes\n",
```

Reply:
Or:
```suggestion
            jl_printf(JL_STDERR, "WARNING: redefinition of constant %s causes undefined behavior, which is not a bug\n",
```

Reply:
This would be printed as

> redefinition of constant `foo` may sometimes fail or cause crashes

and make it sound like it is a particular statement about `foo`.

Reply:
So something like:

> WARNING: redefinition of constant `foo`. Redefining constants may fail or cause crashes.

is better imo.

Reply:
I like that last wording 👍

Reply:
```suggestion
            jl_printf(JL_STDERR, "WARNING: redefinition of constant %s. This may fail, cause incorrect answers, or produce other errors.\n",
```